datafusion/optimizer/src/decorrelate_scalar_subquery.rs

```diff
@@ -173,20 +191,28 @@ fn optimize_scalar(
         .map_err(|e| context!("Exactly one input is expected. Is this a join?", e))?;
     let aggr = Aggregate::try_from_plan(sub_input)
         .map_err(|e| context!("scalar subqueries must aggregate a value", e))?;
-    let filter = Filter::try_from_plan(&aggr.input).map_err(|e| {
-        context!("scalar subqueries must have a filter to be correlated", e)
-    })?;
+    let filter = Filter::try_from_plan(&aggr.input).ok();
+
+    // if there were filters, we use that logical plan, otherwise the plan from the aggregate
+    let input: &LogicalPlan;
```

Comment:
```suggestion
    let input = if filter.is_some() {
        input = &filter.unwrap().input;
    } else {
        input = &aggr.input;
    };
```

Reply:
That won't compile. I googled for how to do this more Rust-like, but didn't find anything more concise that works.

Reply:
Sorry, I was going to pull and compile, but I see you undrafted, so here's what I think will compile and avoid the unwrap() (which we're trying to elliminate for audit purposes even if it should be safe):

```
    let input = match filter {
        Some(filter) => &filter.input,
        None => &aggr.input
    };
    ```

Reply:
Clippy just made me dance, it's all rust-like now :)